src/v/ssx/async-clear.h

```diff
@@ -27,10 +27,11 @@ namespace ssx {
  * Once we are on a version with the fix for #46989 backported,
  * this can be reduced to just a function.
  */
-template<typename K, typename V>
+template<typename K, typename V, typename Hash, typename Eq, typename Alloc>
```

Comment:
This extension is a consequence of using transparent comparators, we need to accept the full template (well, technically not the allocator, but I include it for completeness).